### PR TITLE
Feature/che 781 UI fixes

### PR DESCRIPTION
**Pull request by EvansPie**
CHE-781

# What this PR does

- Fixes button text clipping.
- Change Apple Pay color to black (Apple's default). Discuss if we want this customizable.
- Fix vault manager wrong subtitle copy.
- Fix bug that amount was nullified and amount was not showing on universal checkout.

# Notable decisions & other stuff

n/a

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Core/Primer/Primer.swift [hunk 1]
```diff
@@ -167,12 +167,6 @@ public class Primer {
                 self?.root = RootViewController()
                 guard let root = self?.root else { return }
                 let router: RouterDelegate = DependencyContainer.resolve()
-                let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
-                
-                if flow.internalSessionFlow.vaulted {
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Core/Primer/Primer.swift:172]: What's happening here?

**Review by ceriksson — COMMENTED** [review]
👍 just got some questions regarding dependencies removed + test coverage

File: Sources/PrimerSDK/Classes/Core/Primer/Primer.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Core/Primer/Primer.swift:172]: Those lines were nullifying the amount in the vault flow, leading to a bug if you opened the vault flow and then the checkout flow. We don't need to nullify the amount (as it is one of the variables configured by the dev), we need to not present it in the vault flow.

Comment (ceriksson) [reply Sources/PrimerSDK/Classes/Core/Primer/Primer.swift:172]: makes sense 👍

File: Sources/PrimerSDK/Classes/User Interface/Apple Pay/ApplePayViewModel.swift [hunk 2]
```diff
@@ -57,19 +57,31 @@ class ApplePayViewModel: NSObject, ApplePayViewModelProtocol {
     // swiftlint:disable cyclomatic_complexity function_body_length
     func payWithApple(completion: @escaping (Error?) -> Void) {
         guard let countryCode = countryCode else {
-            return completion(PaymentException.missingCountryCode)
+            let err = PaymentException.missingCountryCode
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/User Interface/Apple Pay/ApplePayViewModel.swift:60]: [Suggestion]
Loads of duplicated code here. Would it be possible to put the common stuff in some function?

File: Sources/PrimerSDK/Classes/User Interface/Checkout/VaultCheckoutViewController.swift [hunk 3]
```diff
@@ -107,7 +107,7 @@ extension VaultCheckoutViewController: UITableViewDelegate, UITableViewDataSourc
             router.root?.dismiss(animated: true, completion: {
                 DispatchQueue.main.async { [weak self] in
                     let appleViewModel: ApplePayViewModelProtocol = DependencyContainer.resolve()
-                    appleViewModel.payWithApple { (_) in
+                    appleViewModel.payWithApple { (err) in
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/User Interface/Checkout/VaultCheckoutViewController.swift:110]: [Q]
We don't do anything with the error? What's the point of this change then?

**Review by xevious78 — APPROVED** [review]
In terms of fixes: it does the job! Just got a couple of non-blocking questions/comments :)

File: Sources/PrimerSDK/Classes/User Interface/Vault/VaultPaymentMethodViewController.swift [hunk 4]
```diff
@@ -22,9 +20,169 @@ internal class VaultPaymentMethodViewController: PrimerViewController {
...
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let viewModel: VaultPaymentMethodViewModelProtocol = DependencyContainer.resolve()
+        
+        if indexPath.row == viewModel.paymentMethods.count {
+            return
+        }
+
+        tableView.deselectRow(at: indexPath, animated: true)
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/User Interface/Vault/VaultPaymentMethodViewController.swift:67]: [Q]
Why do you need to deselect the row when it has been selected?

**Pull request merged**